src/lib/OpenEXRCore/internal_ht.cpp

```diff
@@ -199,16 +199,13 @@ ht_undo_impl (
     cs.read_headers (&infile);
 
     ojph::param_siz siz = cs.access_siz ();
-    ojph::param_nlt nlt = cs.access_nlt ();
 
-    ojph::ui32 image_width =
-        siz.get_image_extent ().x - siz.get_image_offset ().x;
     ojph::ui32 image_height =
         siz.get_image_extent ().y - siz.get_image_offset ().y;
 
     int  bpl       = 0;
     bool is_planar = false;
-    for (ojph::ui32 c = 0; c < decode->channel_count; c++)
+    for (ojph::ui32 c = 0; c < static_cast<ojph::ui32>(decode->channel_count); c++)
```

Comment:
```suggestion
    for (int16_t c = 0; c < encode->channel_count; c++)
```

I would try to avoid the cast altogether if possible.